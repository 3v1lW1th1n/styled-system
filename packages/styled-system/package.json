{
  "name": "styled-system",
  "version": "4.2.2",
  "description": "Responsive, theme-based style props for building design systems with React",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "sideEffects": false,
  "keywords": [
    "react",
    "css-in-js",
    "styled-components",
    "emotion"
  ],
  "author": "Brent Jackson",
  "license": "MIT",
  "devDependencies": {
    "@babel/cli": "^7.2.3"
  },
  "dependencies": {
    "@babel/runtime": "^7.4.2",
    "object-assign": "^4.1.1",
    "prop-types": "^15.7.2"
  },
  "scripts": {
    "prepare": "mkdir -p dist && npm run build:cjs && npm run build:esm",
    "build:cjs": "NODE_ENV=cjs babel src -o dist/index.cjs.js",
    "build:esm": "NODE_ENV=esm babel src -o dist/index.esm.js"
  }
}
