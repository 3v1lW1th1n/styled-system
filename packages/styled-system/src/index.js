export {
  get,
  createParser,
  createStyleFunction,
  style,
  compose,
} from './core'

// split up into separate packages
export { variant } from './variant'
export { themeGet } from './themeGet'
export {
  margin,
  marginTop,
  marginRight,
  marginBottom,
  marginLeft,
  marginX,
  padding,
  paddingTop,
  paddingRight,
  paddingBottom,
  paddingLeft,
  paddingX,
  paddingY,
  space,
  textColor,
  backgroundColor,
  color,
  width,
  fontSize,
  fontFamily,
  fontWeight,
  lineHeight,
  textAlign,
  fontStyle,
  letterSpacing,
  display,
  maxWidth,
  minWidth,
  height,
  maxHeight,
  minHeight,
  size,
  verticalAlign,
  alignItems,
  alignContent,
  justifyItems,
  justifyContent,
  flexWrap,
  flexBasis,
  flexDirection,
  flex,
  justifySelf,
  alignSelf,
  order,
  gridGap,
  gridColumnGap,
  gridRowGap,
  gridColumn,
  gridRow,
  gridAutoFlow,
  gridAutoColumns,
  gridAutoRows,
  gridTemplateColumns,
  gridTemplateRows,
  gridTemplateAreas,
  gridArea,
  border,
  borderWidth,
  borderStyle,
  borderColor,
  borderTop,
  borderRight,
  borderBottom,
  borderLeft,
  borderRadius,
  borders,
  boxShadow,
  opacity,
  overflow,
  background,
  backgroundImage,
  backgroundSize,
  backgroundPosition,
  backgroundRepeat,
  position,
  zIndex,
  top,
  right,
  bottom,
  left,
  buttonStyle,
  textStyle,
  colorStyle,
} from './styles'
