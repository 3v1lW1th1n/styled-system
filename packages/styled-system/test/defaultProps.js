import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import styled from '@emotion/styled'
import { space } from 'styled-system'

expect.addSnapshotSerializer(serializer)

const Comp = styled('div', { shouldForwardProp: () => true })(space)
Comp.defaultProps = { p: 0 }

describe('application of default props should not affect order of styles', () => {
  test('default props not provided to instance', () => {
    const tree = renderer.create(<Comp px={2} py={1} />)
    const props = Object.entries(tree.toJSON().props)

    expect(tree).toMatchSnapshot()
    expect(props).toEqual([
      ['px', 2],
      ['py', 1],
      ['p', 0],
      ['className', 'css-1fhifp9'],
    ])
  })

  test('default props provided to instance', () => {
    const tree = renderer.create(<Comp {...Comp.defaultProps} px={2} py={1} />)
    const props = Object.entries(tree.toJSON().props)

    expect(tree).toMatchSnapshot()
    expect(props).toEqual([
      ['p', 0],
      ['px', 2],
      ['py', 1],
      ['className', 'css-sl57if'],
    ])
  })
})
