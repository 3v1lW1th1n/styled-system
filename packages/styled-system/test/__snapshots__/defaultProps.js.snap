// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application of default props should not affect order of styles default props not provided to instance 1`] = `
.emotion-0 {
  padding: 0;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

<div
  className="emotion-0"
  p={0}
  px={2}
  py={1}
/>
`;

exports[`application of default props should not affect order of styles default props provided to instance 1`] = `
.emotion-0 {
  padding: 0;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

<div
  className="emotion-0"
  p={0}
  px={2}
  py={1}
/>
`;
